invenio_curations/assets/semantic-ui/js/invenio_curations/requests/ModalTriggers.js

```diff
@@ -0,0 +1,173 @@
+  onClick,
+  requestType,
+  loading,
+  ariaAttributes,
+  size,
+  className,
+}) => {
+  const text = i18next.t("Request changes");
+  return (
+    <MediaContextProvider>
+      <Media greaterThanOrEqual="tablet">
+        <Button
+          icon="exclamation circle"
+          labelPosition="left"
+          content={text}
+          onClick={onClick}
+          negative
+          loading={loading}
+          disabled={loading}
+          size={size}
+          className={className}
+          {...ariaAttributes}
+        />
+      </Media>
+      <Media at="mobile">
+        <Dropdown.Item
+          icon={{
+            name: "exclamation circle",
+            color: "negative",
+            className: "mr-5",
+          }}
+          onClick={onClick}
+          content={text}
+        />
+      </Media>
+    </MediaContextProvider>
+  );
+};
+
+RequestCritiqueModalTrigger.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  loading: PropTypes.bool,
+  ariaAttributes: PropTypes.object,
+  size: PropTypes.string,
+  className: PropTypes.string,
+};
+
+RequestCritiqueModalTrigger.defaultProps = {
+  size: "mini",
+  className: "ml-5",
+};
+
+export const RequestResubmitModalTrigger = ({
+  onClick,
+  requestType,
+  loading,
+  ariaAttributes,
+  size,
+  className,
+}) => {
+  const text = i18next.t("Resubmit for review");
+  return (
+    <MediaContextProvider>
+      <Media greaterThanOrEqual="tablet">
+        <Button
+          icon="paper hand outline"
+          labelPosition="left"
+          content={text}
+          onClick={onClick}
+          neutral
+          loading={loading}
+          disabled={loading}
+          size={size}
+          className={className}
+          {...ariaAttributes}
+        />
+      </Media>
+      <Media at="mobile">
+        <Dropdown.Item
+          icon={{
+            name: "paper hand outline",
+            color: "neutral",
+            className: "mr-5",
+          }}
+          onClick={onClick}
+          content={text}
+        />
+      </Media>
+    </MediaContextProvider>
+  );
+};
+
+RequestResubmitModalTrigger.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  loading: PropTypes.bool,
+  ariaAttributes: PropTypes.object,
+  size: PropTypes.string,
+  className: PropTypes.string,
+};
+
+RequestResubmitModalTrigger.defaultProps = {
+  size: "mini",
+  className: "ml-5",
+};
+
+export const RequestReviewModalTrigger = ({
+  onClick,
+  requestType,
+  loading,
+  ariaAttributes,
+  size,
+  className,
+}) => {
+  const text = i18next.t("Start review");
+  return (
+    <MediaContextProvider>
+      <Media greaterThanOrEqual="tablet">
+        <Button
+          icon="eye"
+          labelPosition="left"
+          content={text}
+          onClick={onClick}
+          neutral
+          loading={loading}
+          disabled={loading}
+          size={size}
+          className={className}
+          {...ariaAttributes}
+        />
```

Comment:
Would it make sense to reuse the `RequestReviewButton`? I assume this would remove some duplication and move the styling and text completely to its own component

Same for the other modals and its buttons

Reply:
Very likely yes, I'll try to reduce code duplication!

Reply:
alright, i tried to refactor some of the components – the modal triggers worked pretty well, but for the rest there wasn't a whole lot of saving.
declaring a base button and reusing that for critique, resubmit and review actually slightly increased the lines of code – using JSX would've even further increased the line count for them.
as such, i left the really small components as they were before.

Reply:
oh wow, that is quite some generalization :smile: 
My concern was more about the text and styling of a button. Lets say the `RequestCritiqueButton`. If we want to change the styling, text or icon of the button, we have to change it in two places - which makes it easier to forget about one place. So maybe we can pass the `CritiqueButton` into the modal? Just a thought though - fine with it as it is now as well (until we find some problems down the way)

Reply:
OH, sorry, i misunderstood that one :D 
you mean basically using the the `Button` components in the `ModalTrigger` components?
yeah, that would make sense and should be doable :)